### DISCUSSION TITLE
Request to obtain the sourse code of  Figure 6 "The elastic rod under torsion..." in the article "Direct positon-based solver for stiff rods.

### DISCUSSION THREAD

**Discussion by zhouweihb** [InteractiveComputerGraphics/PositionBasedDynamics#151] [Q&A]
Could you please to upload the code of example KNOT in figure 6 in the rticle "Direct positon-based solver for stiff rods". Thanks very much!
![image](https://user-images.githubusercontent.com/82623815/215315241-e3e01a64-e48b-4bc4-ac0c-ef6d54c9a4ea.png)

**Comment by janbender** [comment]
This example was set up in an old framework which we do not maintain anymore. Unfortunately, nobody had time so far to implement the collisision detection for rods in the PBD framework.

**Reply by fgbxx** [reply]
hi janbender
   Could you share this old framework or the rod collision part for us , very grateful!
Best,
Gang

**Reply by hadar-segal-isi** [reply]
Would it be possible to get a link to the old framework (e.g. commit hash)? I would be happy to try port it to the latest framework.

**Comment by zhouweihb** [comment]
Hi, thanks for you reply!  In the simulation of rope contact, the penetration phenomenon will occur. The time step is adjusted to 0.001, and the contact mode of addCollisionCylinder is used. How to deal with this phenomenon? Please help with your suggestions.
![image](https://user-images.githubusercontent.com/82623815/218126266-7980f998-e2c6-4a66-9930-336810dd271a.png)

**Comment by janbender** [comment]
I assume that the two adjacent collision cylinders will report a contact which is then resolved. In our simulation in the paper we deactivated the collision detection for neighboring cylinder elements. However, at the moment there is no functionality in the PBD library to handle this but I think this should not be hard to implement.

**Reply by hadar-segal-isi** [reply]
Here is my attempt to reproduce the results in the paper:
PR https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/pull/200
